.github/workflows/operator-e2e-integration-tests.yml

```diff
@@ -62,10 +62,44 @@ jobs:
           kubectl get pods --all-namespaces
           kubectl describe nodes
 
+      - name: Clean up docker images
+        if: always()
+        run: |
+          docker images --format '{{.Repository}}:{{.Tag}}' | grep 'feast' | xargs -r docker rmi -f
+          docker system prune -a -f
+
+      - name: Run Previous version tests
+        run: |
+          # Run the previous version tests
+          cd infra/feast-operator/
+          make test-previous-version
+
+      - name: Debug KIND Cluster when there is a failure
```

Comment:
This can be removed. i see it is duplicated.

Reply:
I have not tested but my understanding is this DEBUG step will execute if any one of the step fails in the current job.

Reply:
> I have not tested but my understanding is this DEBUG step will execute if any one of the step fails in the current job.

@lokeshrangineni In GitHub Actions, the if: failure() condition is used to determine whether a step should run only if any previous steps have failed in the workflow.

If multiple steps failed I want to print the cluster state after each step that failed and not only the state after the last failure, that why I don't think we should remove the cleanup steps and leave only one in the end